activerecord/lib/active_record/encryption/encrypted_attribute_type.rb

```diff
@@ -85,11 +85,13 @@ def decrypt(value)
           with_context do
             unless value.nil?
               if @default && @default == value
-                value
+                decrypted_value = value
               else
-                encryptor.decrypt(value, **decryption_options)
+                decrypted_value = encryptor.decrypt(value, **decryption_options)
               end
             end
+
+            reserialize(decrypted_value)
```

Comment:
I'd consider extracting the `unless` statement to its own `#decrypt_as_text(value)` method, and then do:

```ruby
reserialize decrypt_as_text(value) 
```

And, for symmetry, do the same with a `#encrypt_as_text(value)` in #44052 

```ruby
deserialize encrypt_as_text(value)
```

Reply:
I've updated that here - https://github.com/rails/rails/pull/50920/commits/16330f172a9b0ec0396832a58674b1f4d2a2cb50 👍